Add matrix get support

porteri opened the pull request:
This adds support to `get` real data from 2d matrices and arrays of matrices.

To use this `get` feature:
```
real,dimension(:,:) :: foo_2d
real,dimension(:,:,:) :: foo_3d

call json%get('matrix_id',foo_2d, found)
call json%get('matrix_id',foo_3d, found)
```

porteri committed:
WIP: Interface for matrix calls

porteri committed:
WIP: matrix test

porteri committed:
Can get 2d matrix and array of matrices

porteri committed:
Removed write statement

porteri committed:
Added ability to determine supplied array lengths

An optional argument was added that provides the # of values from the
json file for each row within the matrix. This allows the user to more
easily know if not all of the rows had the same number of columns.

porteri commented:
...I am by NO means saying this is elegant or the best way to have performed this...

zbeekman committed:
Print test output on failure with check target

zbeekman commented on files/inputs/test36.json:
```diff
@@ -0,0 +1,38 @@
+{
+"fooList": [
+	[
+	    [ 0.2,0.01,0.02,0.04],
```
Are you intentionally wrapping the matrix (array of arrays) in an additional outer array (with only one element, the matrix in question)?

porteri replied:
Hmm, nope! Looks like a copy error on my part. I'll fix this and test to see if that's the issue

zbeekman commented on files/inputs/test36.json:
```diff
@@ -0,0 +1,38 @@
+		[ 50.1,30.2],
+		[ 50.1,30.2,0.01,0.02,0.04],
+		[ 0.2,0.01,0.02],
+		[ 0.2,0.01,0.02,0.04]
```
@porteri I have not looked into this in detail yet, however, my impression was that JSON-Fortran's extant matrix capabilities do not support sparse/ragged edge matrices. Is this a requirement for you? It will likely be easier to implement M by N matrices than ragged edge/sparse matrices in the JSON read & write routines.

porteri replied:
Yea unfortunately it is a requirement. That's why I added in the optional `matrix_column_size` flag. It works fine reading b/c I allocate it to the largest size of N (currently hardwired to 1000 and then reduced but will fix this in future)

porteri committed:
WIP: Found issue with new logic

porteri committed:
Write format issue fix

zbeekman committed:
Fix bad JSON in test

zbeekman committed:
Merge branch 'master' into add_matrix_support

zbeekman committed:
Merge branch 'master' into add_matrix_support

 - Update with latest upstream enhancements 6.9.0 --> 6.10.0

Commit:
Fix for undefined variable in test#26

Fixes the undefined variable error_cnt for test #26 when the json%validate returns a true is_valid statement.

porteri committed:
Merge pull request #2 from nrc-fuels/fix_failing_test

Fix for undefined variable in test#26

Commit:
Merge branch 'master' into add_matrix_support

Commit:
Merge fix

zbeekman committed:
Merge branch 'master' of github.com:jacobwilliams/json-fortran into add_matrix_support

zbeekman committed:
Only install module files, not config dir with IDE

 - MSVS will install, e.g., `$<prefix>/include/Debug/json_file_module.mod`
 - We should strip the build config directory that IDEs like MSVS add when doing an installation

zbeekman committed:
Merge branch 'module_install_w_IDEs' of github.com:zbeekman/json-fortran into add_matrix_support

zbeekman committed:
Revert "Merge branch 'module_install_w_IDEs' of github.com:zbeekman/json-fortran into add_matrix_support"

This reverts commit ce9c43b39963730d91792b1f6d4f411b1a58719d, reversing
changes made to 34a446e7457d9b3845d67a784404ae646fd5a81c.

zbeekman committed:
Merge branch 'module_install_w_IDEs' into add_matrix_support

zbeekman committed:
Add option to build and link against OpenCoarrays

zbeekman committed:
Merge tag '7.1.0' of github.com:jacobwilliams/json-fortran into add_matrix_support

zbeekman committed:
Move matrix test (40.f90) to 41.F90

zbeekman committed:
Testing: Use CTest fixtures to manipulate files

 - Fixes #413
 - Use fixtures to delete old json outputs before re-running tests, then copy
   pristine inputs back into the build directory

zbeekman committed:
TESTING: Use fixtures to copy and cleanup files

 - Fixes #413
 - Also fixes regression testing and output validation

zbeekman committed:
Add a config option to compile & link against OC

 - Passing `-fcoarray=lib` to gfortran causes the compiler to generate
   different modules & symbols. This is likely a bug in gfortran, but,
   a consequence is that you need to build libraries you link to with the
   same `-fcoarray=...` flag.

zbeekman committed:
Travis-CI: Install jsonlint from node/npm

zbeekman committed:
CMake: Don't rely on diff for testing

jacobwilliams committed:
fixed a dangling pointer in test 10. Fixes #422

jacobwilliams committed:
Merge pull request #423 from jacobwilliams/422-dangling-pointer

fixed a dangling pointer in test 10. Fixes #422

zbeekman committed:
CHANGELOG: Add entry/comment about OpenCoarrays

jacobwilliams committed:
Merge pull request #421 from nrc-fuels/test-fixtures-413

Test fixtures 413

zbeekman committed:
Merge branch 'master' of github.com:jacobwilliams/json-fortran into add_matrix_support

zbeekman committed:
Fix matrix test name clashes

zbeekman committed:
CMake: Remove extra instance of OC linking option

zbeekman committed:
Allow ragged edge matrices in `matrix_info`s

 - Add additional optional, intent out parameter, `is_uniform`
   - `.true.` for regular matrices
   - `.false.` for ragged edge matrices
 - The idea is to allow trailing zeros to be compressed/elided (eventually)
 - The `set_size` dummy argument was renamed to `mx_set_size` and records the
   largest column size, assuming __*row* major order__

zbeekman committed:
WIP: Transfering from laptop to work machine

zbeekman committed:
WIP: Commit unsaved changes to move from laptop

zbeekman committed:
Remove errant string matrix overloads/interfaces

dcolameco committed:
Updating to remove json tests for Windows as a temporary workaround before json is removed from FAST
